olive/evaluator/evaluation.py

```diff
@@ -301,15 +301,16 @@ def get_user_config(config: dict):
     dataloader = None
     dataloader_func = getattr(config, "dataloader_func", None)
     dataloader = user_module.call_object(dataloader_func, config.data_dir, config.batch_size)
-    if not dataloader:
-        dataloader = DummyDataloader(config.input_names, config.input_shapes, config.input_types)
 
     post_func = getattr(config, "post_processing_func", None)
     post_func = user_module.load_object(post_func)
 
     eval_func = getattr(config, "evaluate_func", None)
     eval_func = user_module.load_object(eval_func)
 
+    if not dataloader and not eval_func:
```

Comment:
![image](https://user-images.githubusercontent.com/13343117/232727935-45291cc4-809e-4c44-90dd-2dea91247cb3.png)

Emm, in somecase, dataloader is a function/object, `not dataloader` failed with above exceptions. Maybe `is not None` is ok here.

Reply:
Failed from user_script where forget to set dataset length.